src/Streamly/Coreutils/FileTest.hs

```diff
@@ -322,3 +361,67 @@ isOlderThan = compareModTime (<)
 -- Like coreutil @test file1 -ef file2@.
 isSameFile :: FilePath -> IO FileTest
 isSameFile = undefined
+
+data TimeStampMin = TimeStampMin Ordering Int
+
+data FileSize = FileSize Ordering Integer
+
+getLocalTime :: IO MilliSecond64
+getLocalTime = fromAbsTime <$> getTime Realtime
+
+compareFsTime :: (FileStatus -> EpochTime) -> TimeStampMin -> IO FileTest
+compareFsTime toEpoc fopt = do
+    t <- getLocalTime
+    return $ FileTest (Predicate ( `f` t))
+
+    where
+
+    f st t =
+        case toEpoc st of
+            CTime t1 ->
+                let t2 =
+                        case fopt of
+                            TimeStampMin LT n ->
+                                t <
+                                MilliSecond64
+                                    (t1 * 1000 + fromIntegral (n * 60000))
+                            TimeStampMin GT n ->
+                                t >
+                                MilliSecond64
+                                    (t1 * 1000 + fromIntegral (n * 60000))
+                            TimeStampMin EQ n ->
+                                t ==
+                                MilliSecond64
+                                    (t1 * 1000 + fromIntegral (n * 60000))
+                  in t2
+
+compareAccessTime :: TimeStampMin -> IO FileTest
+compareAccessTime = compareFsTime Files.accessTime
```

Comment:
This API is of very limited use, it compares the current time with the file timestamps. When would a file time be in future, wouldn't it almost always be less than the current time?

A better API would be to compare a particular time with the file's timestamp.

The name of the API should be intuitive.

Reply:
Rather make a `compareAccessTime :: (POSIXTime -> POSIXTime -> Bool) -> FilePath -> IO FileTest`. And then more specialized APIs could be written as:

`isNewerThanTime :: POSIXTime -> IO FileTest`
`isOlderThanTime :: POSIXTime -> IO FileTest`